README.md

```diff
@@ -1,151 +1,154 @@
-const senderAddress = '0xfceA770875E7e6f25E33CEa5188d12Ef234606b4';
-const receiver = '0x8B4e846c90a2521F0D2733EaCb56760209EAd51A'; // Useful in case of swap and transfer
-const referrer = 'my-company-or-nick-name';
+const srcDecimals = 18
+const srcAmount = '1000000000000000000'; // The source amount multiplied by its decimals
+const destToken = '0xcAfE001067cDEF266AfB7Eb5A286dCFD277f3dE5';
+const destDecimals = 18
+const destAmount = priceRoute.destAmount // price route being output of paraSwap.getRate()
+const senderAddress = '__sender_address__'; // mandatory
+const receiver = '__receiver_address__'; // optional: for swap and transfer
+const partnerAddress = '__fee_receiver_address__' // optional: for permission-less monetization
+const partnerFeeBps = 50 // optional: fee in base point, for permission-less monetization
+
 
 const txParams = await paraSwap.buildTx(
-  srcToken,
-  destToken,
-  srcAmount,
-  destAmount,
-  priceRoute,
-  senderAddress,
-  referrer,
-  receiver,
+  {
+    srcAmount,
+    srcToken,
+    srcDecimals,
+    destAmount,
+    destToken,
+    destDecimals,
+    priceRoute,
+    senderAddress,
+    receiver,
+    partnerAddress,
+    partnerFeeBps,
+  }
 );
 
-web3.eth.sendTransaction(
-  txParams,
-  async (err: Error, transactionHash: string) => {
-    if (err) {
-      return this.setState({ error: err.toString(), loading: false });
-    }
-    console.log('transactionHash', transactionHash);
-  },
-);
+const transactionResponse = await signer.sendTransaction(txParams);
+const transactionReceipt = await transactionResponse.wait();
 ```
 
----
+## Bundle Optimization
+For bundle-size savvy developers, you can construct a lightweight version of this sdk to bring only the functions you need.
 
-### To run the example locally:
+For e.g. for fetching rate and allowance only
 
-Created an .env file with these 2 env variables:
+```typescript
+import { constructPartialSDK, constructFetchFetcher, constructGetRate, constructGetAllowance } from '@paraswap/sdk';
 
-```bash
-PROVIDER_URL=YOUR_PROVIDRER_URL_OR_INFURA_URL
-NODE_ENV=production
-```
+const fetcher = constructFetchFetcher(window.fetch)
 
-run
+const minParaSwap = constructPartialSDK({
+  network: 1,
+  fetcher,
+}, constructGetRate, constructGetAllowance)
 
-```bash
-yarn install paraswap
+const priceRoute = await minParaSwap.getRate(params)
+const allowance = await minParaSwap.getAllowance(userAddress, tokenAddress);
 ```
 
-For local developement you can run
+## Legacy
+The ParaSwap class is exposed with some effort on backward compatibility with previous versions.
 
-```bash
-yarn dev
-```
 
-For production build:
+```typescript
+import { ParaSwap } from '@paraswap/sdk'
+import axios from 'axios'
+import Web3 from 'web3'
+
+const web3Provider = new Web3(window.ethereum)
+const account = '__user_address__'
+
+const paraswap = new ParaSwap(
+  1, 
+  undefined, 
+  web3Provider, 
+  null, 
+  account, 
+  axios
+)
```

Comment:
Please add a similar example without providers and without axios, as that is what most people want to get rid of.
Like the one above for `Bundle Optimization` but with legacy class

Reply:
done